### ISSUE TITLE
Add ARIA roles and attributes to Dropdown component

### ISSUE THREAD

**Issue by ianoxley** [Sage/carbon#1471]
The `Dropdown` component is missing the `combobox` ARIA role and does not set the required ARIA attributes on its child elements as desribed on https://www.w3.org/TR/wai-aria-1.1/#combobox

> A composite widget containing a single-line textbox and another element, such as a listbox or grid, that can dynamically pop up to help the user set the value of the textbox.
> 
> Authors must ensure an element with role combobox contains or owns a text input element with role textbox or searchbox and that the text input has aria-multiline set to false. If the combobox provides autocompletion behavior for the text input as described in aria-autocomplete, authors must set aria-autocomplete on the textbox element to the value that corresponds to the provided behavior.
> 
> Typically, the default state of a combobox is collapsed. In the collapsed state, only the textbox element of a combobox is visible. A combobox is said to be expanded when both the textbox and a secondary element that serves as its popup are visible. Authors must set aria-expanded to true on an element with role combobox when it is expanded and false when it is collapsed. Elements with the role combobox have an implicit aria-expanded value of false.
> 
> When a combobox is expanded, authors must ensure it contains or owns an element that has a role of listbox, tree, grid, or dialog. This element is the combobox popup. When the combobox is expanded, authors must set aria-controls on the textbox element to a value that refers to the combobox popup element.
> 
> Elements with the role combobox have an implicit aria-haspopup value of listbox. If the combobox popup element has a role other than listbox, authors must specify a value for aria-haspopup that corresponds to the type of its popup.
> 
> To be keyboard accessible, authors should manage focus of descendants for all instances of this role, as described in Managing Focus. When a combobox receives focus, authors should ensure focus is placed on the textbox element.
> 
> Authors should provide keyboard mechanisms for moving focus between the textbox element and the elements contained in the popup. For example, one common convention is that Down Arrow moves focus from the text input to the first focusable descendant of the popup element. If the popup element supports aria-activedescendant, in lieu of moving focus, such keyboard mechanisms can control the value of aria-activedescendant on the textbox element. When a descendant of the popup element is active, authors may set aria-activedescendant on the textbox to a value that refers to the active element within the popup while focus remains on the textbox element.

The list that displays the available options should have the `listbox` ARIA role and associated atrributes that are outlined on https://www.w3.org/TR/wai-aria-practices-1.1/#Listbox

> **WAI-ARIA Roles, States, and Properties**
> 
> - An element that contains or owns all the listbox options has role listbox.
> - Each option in the listbox has role option and is a DOM descendant of the element with role listbox or is referenced by an aria-owns property on the listbox element.
> - If the listbox is not part of another widget, then it has a visible label referenced by aria-labelledby on the element with role listbox.
> - In a single-select listbox, the selected option has aria-selected set to true.
> - if the listbox supports multiple selection:
> -- The element with role listbox has aria-multiselectable set to true.
> -- All selected options have aria-selected set to true.
> -- All options that are not selected have aria-selected set to false.
> - If the complete set of available options is not present in the DOM due to dynamic loading as the user scrolls, their aria-setsize and aria-posinset attributes are set appropriately.
> - If options are arranged horizontally, the element with role listbox has aria-orientation set to horizontal. The default value of aria-orientation for listbox is vertical.

## Related issue 
* should be cross QA'd: https://github.com/Sage/carbon/issues/1525

**Comment by ianforster-sage** [comment]
Accessibility improvements are on our roadmap.
